cell_classification/segmentation_data_prep_test.py

```diff
@@ -142,15 +162,68 @@ def test_check_input():
 
         # check if the normalization_dict is calculated in check_input when
         # data_folders but no normalization_dict_path is given to init
-        data_prep = prep_object(
-            data_folders=data_folders,
-            conversion_matrix_path=conversion_matrix_path,
-            tf_record_path=os.path.join(temp_dir, "tf_record_path"),
-            cell_table_path=cell_table_path,
-        )
+        data_prep.data_folders = data_folders
+        data_prep.normalization_dict_path = None
         data_prep.check_input()
         assert norm_dict == data_prep.normalization_dict
 
+        # check if ValueError is raised if selected_markers in conversion_matrix
+        # but not in loaded normalization_dict
+        conversion_matrix = pd.DataFrame(
+            np.random.randint(0, 2, size=(6, 6)),
+            columns=["CD11c", "CD14", "CD56", "CD57", "XYZ", "ZYX"],
+            index=["stromal", "FAP", "NK", "CD4T", "CD14", "CD163"],
+        )
+        conversion_matrix_path = os.path.join(temp_dir, "conversion_matrix.csv")
+        conversion_matrix.to_csv(conversion_matrix_path, index=False)
+        data_prep = copy.deepcopy(data_prep_working)
+        data_prep.conversion_matrix_path = conversion_matrix_path
+        data_prep.normalization_dict_path = os.path.join(temp_dir, "norm_dict.json")
+        data_prep.selected_markers = ["ZYX"]
+        with pytest.raises(ValueError):
+            data_prep.check_input()
+
+        # check if FileNotFoundError is raised if data_folders and conversion_matrix_path are given
+        # together with selected_markers were images are missing for in data_folders
+        data_prep = copy.deepcopy(data_prep_working)
```

Comment:
This is a very indirect way of finding out that the image data does not match up with the specified markers. Better to find out earlier. 

In addition, we generally only want to check the behavior of the function directly under test. This checks the behavior of a function that check_input is calling and the fucntion that function is calling etc. 

Just get rid of this one